src/mui/input/FileInputPreview.js

```diff
@@ -1,65 +1,72 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import FlatButton from 'material-ui/FlatButton';
+import IconButton from 'material-ui/IconButton';
 import { pinkA200 } from 'material-ui/styles/colors';
 import RemoveCircle from 'material-ui/svg-icons/content/remove-circle';
 
 const styles = {
-    container: {
-        display: 'inline-block',
-        position: 'relative',
-    },
-    removeButton: {
-        position: 'absolute',
-        top: '0.5rem',
-        right: '0.5rem',
-        minWidth: '2rem',
-        opacity: 0,
-    },
     removeButtonHovered: {
         opacity: 1,
     },
 };
 
-export class ImageInputPreview extends Component {
+export class FileInputPreview extends Component {
     constructor(props) {
         super(props);
         this.state = {
             hovered: false,
         };
     }
 
+    componentWillUnmount() {
+        const { file } = this.props;
+
+        if (file && file.preview && window && window.URL && window.URL.revokeObjectURL) {
```

Comment:
I think we can prevent of all these checks as support seems quite nice: https://developer.mozilla.org/fr/docs/Web/API/URL/revokeObjectURL